azure-client-runtime/src/main/java/com/microsoft/azure/CloudError.java

```diff
@@ -33,6 +35,11 @@
      */
     private List<CloudError> details;
 
+    /**
+     * The inner error.
+     */
+    private ObjectNode innererror;
```

Comment:
What's the format of the innerError? Will it be a HashMap? (judging on your test)

Also please camel case the variables and methods.